src/components/ConnectionPage/Dashboard.tsx

```diff
@@ -0,0 +1,178 @@
+        requestId: req.connectionId, // Ensure this matches what the PendingRequests component expects
+        id: req.requester.accountId,
+        username: req.requester.username,
+      }));
+      setPendingRequests(mappedRequests);
+    } catch (error) {
+      console.error('Error fetching pending requests:', error);
+    }
+  };
+
+  const fetchMyConnections = async () => {
+    try {
+      const data = await findConnectionsByUserId(userId);
+      const mappedConnections = data.map((conn: any) => ({
+        connectionId: conn.connectionId,
+        id: conn.recipient.accountId !== userId
+          ? conn.recipient.accountId
+          : conn.requester.accountId,
+        username: conn.recipient.accountId !== userId
+          ? conn.recipient.username
+          : conn.requester.username,
+      }));
+      setMyConnections(mappedConnections);
+    } catch (error) {
+      console.error('Error fetching connections:', error);
+    }
+  };
+
+  const handleSendConnectionRequest = async (recipientId: number) => {
+    try {
+      if (recipientId !== userId) {
+        await sendConnectionRequest(userId, recipientId);
+        const recipientName = searchResults.find(user => user.id === recipientId)?.username;
+        toast.success(`You have sent a connection request to ${recipientName}!`);
+        await fetchPendingRequests();
+      } else {
+        toast.error("You cannot send a connection request to yourself.");
+      }
+    } catch (error) {
+      toast.error('Error sending connection request.');
+      console.error('Error sending connection request:', error);
+    }
+  };
+
+  const handleAcceptRequest = async (requestId?: number) => {
+    if (requestId) {
+      try {
+        await acceptConnectionRequest(requestId);
+        toast.success('Connection request accepted!');
+        await fetchPendingRequests();
+        await fetchMyConnections();
+      } catch (error) {
+        toast.error('Error accepting request.');
+        console.error('Error accepting request:', error);
+      }
+    }
+  };
+
+  const handleRejectRequest = async (requestId?: number) => {
+    if (requestId) {
+      try {
+        await rejectConnectionRequest(requestId);
+        toast.info('Connection request rejected.');
+        await fetchPendingRequests();
+      } catch (error) {
+        toast.error('Error rejecting request.');
+        console.error('Error rejecting request:', error);
+      }
+    }
+  };
+
+  const handleRemoveConnection = async (connectionId?: number) => {
+    if (connectionId) {
+      try {
+        await removeConnection(connectionId);
+        toast.info('Connection removed.');
+        await fetchMyConnections();
+      } catch (error) {
+        toast.error('Error removing connection.');
+        console.error('Error removing connection:', error);
+      }
+    }
+  };
+
+  return (
+    <div className="dashboard">
+      <ToastContainer />
+      <Header />
+      <main className="main-content">
```

Comment:
We don't have to use semantic tags. There may be another `<main>` element on the page at the same time.